### DISCUSSION TITLE
Theme sync with daytime/system theme

### DISCUSSION THREAD

**Discussion by sbatial** [kovidgoyal/kitty#3454] [Q&A] [answered]
Is it currently planned to implement the possibility to automatically change the theme of the theme based on the time of the day or the system settings?
So that one could have a light theme at day and a dark theme at night.
Or is there some workaround I didn't find?

**Comment by kovidgoyal** [comment, marked as answer]
No plans. You can simply write your script to do it. kitty is fully scriptable. See the FAQ

**Reply by sbatial** [reply]
Ok, thank you. 👍

**Comment by magicznyleszek** [comment]
@BodeSpezial were you able to write the script? :)

**Reply by magicznyleszek** [reply]
For future archeologists - I have a script that is ok-ish (it went through several revisions, but is not as good as I would like it to be), so I will share it:

```bash
#!/bin/bash

dark_theme_name="Rosé Pine Moon"
light_theme_name="Rosé Pine Dawn"

function toggle_kitty_theme {
  # This either returns "Dark" or it doesn't exist (i.e. "Light")
  is_os_dark=$(defaults read -g AppleInterfaceStyle)

  theme_to_set="$light_theme_name"
  if [[ $is_os_dark = 'Dark' ]]; then
    theme_to_set="$dark_theme_name"
  fi

  if [[ $theme_to_set = $dark_theme_name ]]; then
    echo "Applying dark kitty theme…"
  else 
    echo "Applying light kitty theme…"
  fi

  # setting kitty theme and force reloading it
  kitty @ kitten themes --reload-in=all "$theme_to_set"
  for kittyid in $(ps -e | grep "[k]itty" | awk '{print $1}' | fmt); do
    kill -s USR1 $kittyid
  done
}

toggle_kitty_theme
```

It does work, but have some caveats:
1. Sometimes MacOS requires you to run this twice; I haven't come up with why it happens, it has something to do with the `kittyid` 🤷
2. You need to run it manually 👎

**Reply by ZuBB** [reply]
> `ps -e | grep "[k]itty"`

`pgrep kitty`

**Reply by kovidgoyal** [reply]
You dont need to send USR1 to kitty at all, the themes kitten with --reload-all does that already.

**Reply by 0rphee** [reply]
I used this https://github.com/bouk/dark-mode-notify

It does the job very well.

**Comment by gregorias** [comment]
[Theme change based on OS settings is now possible since 0.38.](https://sw.kovidgoyal.net/kitty/changelog/#id2:~:text=Allow%20specifying%20individual%20color%20themes%20to%20use%20so%20that%20kitty%20changes%20colors%20automatically%20following%20the%20OS%20dark/light%20mode)

**Reply by olejorgenb** [reply]
In gnome 47.9 with Kitty 0.42.2 it does not work fully. It detected a change from light to dark in existing terminals, but not for dark to light.

**Reply by olejorgenb** [reply]
I suppose it has to do with the note here: https://sw.kovidgoyal.net/kitty/kittens/themes/#change-color-themes-automatically-when-the-os-switches-between-light-and-dark

> On the GNOME desktop, the desktop reports the color preference as no-preference when the “Dark style” is not enabled. So use no-preference-theme.auto.conf to select colors for light mode on GNOME. You can manually enable light style with gsettings set org.gnome.desktop.interface color-scheme prefer-light in which case GNOME will report the color scheme as light and kitty will use light-theme.auto.conf.
